service/store/store.go

```diff
@@ -505,11 +510,18 @@ func (s *Store) WriteDealData(b *Bid) (string, error) {
 
 // WriteDataURI writes the uri resource to the configured deal data directory.
 func (s *Store) WriteDataURI(bidID auction.BidID, payloadCid, uri string, size uint64) (string, error) {
-	duri, err := datauri.NewURI(payloadCid, uri)
-	if err != nil {
-		return "", fmt.Errorf("parsing data uri: %w", err)
+	carDownloadPath := s.dealDataFilePathFor(bidID, payloadCid)
+
+	if s.boostedDownload {
+		ok := s.boostDownload(s.ctx, uri, carDownloadPath)
+		if ok {
+			return carDownloadPath, nil
+		}
 	}
```

Comment:
As an experimental feature, trying to keep it separate.
If the flag has been enabled, we try using this download style. If for any reason things didn't worked out, we fallback with the usual style.

What's outside this if case is the normal download style.